pkg/blockstorage/azure/azuredisk.go

```diff
@@ -489,6 +489,13 @@ func (s *adStorage) VolumeCreateFromSnapshot(ctx context.Context, snapshot block
 	if id != "" {
 		createDisk.Zones = azto.StringSlicePtr([]string{id})
 	}
+	for _, saType := range azcompute.PossibleDiskStorageAccountTypesValues() {
```

Comment:
@bathina2 - what's the behavior if we don't find a match - do we need to return an error?

Reply:
The default is to take the type from the from the snapshot, which is what it was doing originally. You can pass it a sku.Name type and it will change the resulting volume to the new type.

Reply:
I made it a lenient check. Unless you prefer I make it strict?